sdk/client-core/core/src/main/java/io/clientcore/core/http/RestProxy.java

```diff
@@ -56,8 +56,8 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         final SwaggerMethodParser methodParser = getMethodParser(method);
         RequestOptions options = methodParser.setRequestOptions(args);
 
-        return restProxyImpl.invoke(proxy, method, options, options != null ? options.getRequestCallback() : null,
-            methodParser, args);
+        return restProxyImpl.invoke(proxy, method, options, (options != null && options != RequestOptions.NONE)
+            ? options.getRequestCallback() : null, methodParser, args);
```

Comment:
I'm not sure about this - I don't think I like the pattern of having to check throughout our code for `RequestOptions.NONE` (as it feels error-prone). Can we not have this check done inside the `RequestOptions` type itself (i.e. `if (this == RequestOptions.NONE) return null`)

Reply:
Oh - I see why this is like this - because your getter method throws an exception. As I say further down, we should tweak this.

Reply:
I think I stuck with the getters throwing because some of these properties are mutable. We do some special handling for RequestOptions.NONE like you suggested in another comment.